controlplane/src/core/workers/DeleteUserQueue.ts

```diff
@@ -0,0 +1,141 @@
+import { PlatformEventName } from '@wundergraph/cosmo-connect/dist/notifications/events_pb';
+import { ConnectionOptions, Job, JobsOptions, Queue, Worker } from 'bullmq';
+import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
+import pino from 'pino';
+import * as schema from '../../db/schema.js';
+import { BlobStorage } from '../blobstorage/index.js';
+import { OrganizationRepository } from '../repositories/OrganizationRepository.js';
+import { UserRepository } from '../repositories/UserRepository.js';
+import Keycloak from '../services/Keycloak.js';
+import { PlatformWebhookService } from '../webhooks/PlatformWebhookService.js';
+
+const QueueName = 'user.delete';
+const WorkerName = 'DeleteUserWorker';
+
+export interface DeleteUserInput {
+  userId: string;
+}
+
+export class DeleteUserQueue {
+  private readonly queue: Queue<DeleteUserInput>;
+  private readonly logger: pino.Logger;
+
+  constructor(log: pino.Logger, conn: ConnectionOptions) {
+    this.logger = log.child({ queue: QueueName });
+    this.queue = new Queue<DeleteUserInput>(QueueName, {
+      connection: conn,
+      defaultJobOptions: {
+        removeOnComplete: true,
```

Comment:
We should not delete it and keep a history for more than 90d.

Reply:
Done